### PR TITLE
Feature/resource proxy rewrite

### PR DESCRIPTION
**Pull request by Mechazawa**
- Minor cleanups
 - Remove cache time parameter, it's a property now
 - Update some docstrings
 - fix `listWrapper.on` behaviour
 - `ResourceCache::_timestamp` is a getter now
 - Remove `search` and `searchAndWrap` from `SimpleResourceProxy`
 - `SimpleResourceProxy::list*` requires an object containing it's parameters now

### CODE REVIEW

**Comment by Mechazawa** [comment]
Breaking changes!

File: src/SimpleResourceProxy.js [hunk 1]
```diff
@@ -102,91 +101,93 @@ export default class SimpleResourceProxy {
   }
 
   /**
-   * Lists target resource
-   * @param {Object<String, String|Array<String>>} query - Query
-   * @param {Number} page - The page to be requested
-   * @param {Number} perPage - Amount of items per page. This is silently capped by the API
+   * Build a new isntance of the target
```
Comment (beschoenen) [inline src/SimpleResourceProxy.js]: Typo instance

File: src/SimpleResourceProxy.js [hunk 2]
```diff
@@ -102,91 +101,93 @@ export default class SimpleResourceProxy {
...
+  new(data = {}) {
+    // Merge but don't overwrite using seed data
+    data = Object.assign(this._seedData, data);
+
+    return new this.Target(this._api, data);
+  }
+
+  /**
+   * List target resource
+   * @param {Number|Object} [params] - ParametersThe page to be requested
```
Comment (beschoenen) [inline src/SimpleResourceProxy.js]: Typo?

File: src/SimpleResourceProxy.js [hunk 3]
```diff
@@ -102,91 +101,93 @@ export default class SimpleResourceProxy {
...
-   * Lists target resource
-   * @param {Object<String, String|Array<String>>} query - Query
-   * @param {Number} page - The page to be requested
-   * @param {Number} perPage - Amount of items per page. This is silently capped by the API
-   * @param {Number} cacheTime - Amount of seconds to store a value in cache
-   * @param {Boolean} shareCache - Share cache across instances
-   * @returns {PaginatedResourceWrapper} - Wrapped {@link PaginatedResourceListing} instance
-   *
+   * List target resource
+   * @param {Number|Object} [params] - ParametersThe page to be requested
```
Comment (beschoenen) [inline src/SimpleResourceProxy.js]: Typo?

**Pull request merged**